python/ray/data/tests/test_consumption.py

```diff
@@ -158,6 +158,19 @@ def test_count_edge_case(ray_start_regular):
     assert actual_count == 5
 
 
+def test_count_after_partial_execution(ray_start_regular):
+    paths = ["example://iris.csv"] * 5
+    ds = ray.data.read_csv(paths, override_num_blocks=15)
+    for batch in ds.iter_batches(batch_size=1):
+        # Take one batch and break to simulate partial iteration/execution.
+        break
+    # Row count should be unknown after partial execution.
+    assert "num_rows=?" in str(ds)
+    # After calling `ds.count()`, row count should be known.
+    assert ds.count() == 150 * 5
```

Comment:
Nit: To avoid making assumptions about the implementation of count (i.e., that it calls `iter_internal_ref_bundles`), might be better to fully iterate over `iter_internal_ref_bundles` than calling `count`.